apps/bond/components/BondingProducts/Bonding/TokenManagement/hooks/useWhirlpool.jsx

```diff
@@ -75,17 +75,10 @@ const whirlpoolQuery = async () => {
 };
 
 const useWhirlpoolQuery = () => {
-  const [queryResult, setQueryResult] = useState(null);
-
-  useEffect(() => {
-    const fetchData = async () => {
-      const result = await whirlpoolQuery();
-      setQueryResult(result);
-    };
-    fetchData();
+  return useCallback(async () => {
+    const result = await whirlpoolQuery();
+    return result;
   }, []);
-
-  return queryResult;
 };
```

Comment:
is this hook necessary?
there's no lifecycle state stuff going on here any more, we can just call `await whirlpoolQuery()` where the value is required